doc/users/project/mission.rst

```diff
@@ -0,0 +1,28 @@
+Matplotlib Mission Statement
+============================
+
+The Matplotlib developer community develops, maintains, and supports Matplotlib
+and its extensions to provide plotting tools for the Scientific Python
+Ecosystem.  We judge our success based on how effectively we enable our users, across
+scientific, industrial, commercial, and educational domains, to achieve their
```

Comment:
```suggestion
academic, industrial, commercial, and educational domains, to achieve their
```

@noatamir pointed out that "scientific" is the odd-one-out in this sentence and "academic" is a better word for what is meant.

Reply:
The suggested change leaves out NASA, NOAA, USGS, etc.  Is there a concise way to include them?  Or are we trying to do too much enumeration of specific categories?  Maybe just leave out the whole list, and leave it at "enable our users to achieve their goals".  That leaves the domain specification to "Scientific Python Ecosystem"; maybe that is enough.

Reply:
I'm also not sure in that context what the difference between "academic" and "educational" is.

Reply:
As someone who sits in a institution that is in that categories (in the US they tend to be called Federally Funded Research and Development Centers (FFRDC)), we are used to the ambiguity and will self-identify as whichever of the other categories suits us best at the time ;)  We could reference FFRDC but that would miss our peer institutions in other countries (and is _super_ jargonny).

There is some aspect of on-label and off-label uses of Matplotlib.  Our heritage is for scientific plotting (going back to John's EKG plots) and a majority of our developers are trained in some sort of science, and a bulk of the (maybe over) specialized plotting we have is driven by the needs of scientific visualization.  That we have also been used by a (wide) variety of people outside of the domains that our developers work in is a testament to the library and is great!  One of the best things about open source is that our users do things with the code that the core developers could not have imagined.  However, we can not support all domains equally; part of it is just the expertise we have in the core team (I think mpl-finance is a good example here that is _much_ better off with Daniel as a stand-alone project) and because there may be conflicting needs/expectations and some use is should be considered "intended" and some "we are happy it works".

Some of this is very easy to do: "please implement an email client" -> "no, obviously out of scope".  

Some is a bit harder, but still easy: "This API makes it hard to make (intentional) art with Matplotlib " -> "cool you are making art, but the purpose is to be a plotting library so we will not change the API"

Some are super subtle : "if I drop integer coded catagoricals into imshow and use interpolation I get fractional values" -> well, that is a bit of a miss-use of `imshow` because it implicitly assumes the values are continuous not discrete.  do we need a new function, make imshow (yet more) complex,  or is no-norm the solution here?

That is all a very long winded way of saying I think that it is important that we scope the intended application down from "anyone who installs the library and has a story about how they want to use it".

Another aspect that I want to make sure we do not lose is a sense that Matplotlib is always (well intended to be) used as a means to and end, not and end it self.   There is some reason (be it understanding some science measurement, scheduling operations, looking at revenue over time, looking at survey results, looking at market data, ....) that the user is making a plot.  Our goal is not to be the perfect platonic plotting library, but rather to be a tool in service of whatever it is our users are actually doing (I also think that this is the case for a vast majority of software).

Put another, to paraphrase Charity Majors "~9s~ design does not matter if users are not happy" .  We can not make all users happy all of the time so we should identify which users / applications we intend to make happy to help us with triage / decision making.

Reply:
"enable our users to solve problems in their scientific, educational, and operational domains."